crates/burn-autodiff/src/ops/tensor.rs

```diff
@@ -2437,6 +2437,81 @@ impl<B: Backend, C: CheckpointStrategy> FloatTensorOps<Self> for Autodiff<B, C>
             .stateless(B::float_sign(tensor.primitive))
     }
 
+    fn float_expand<const D1: usize, const D2: usize>(
+        tensor: FloatTensor<Self, D1>,
+        shape: Shape<D2>,
+    ) -> FloatTensor<Self, D2> {
+        #[derive(Debug)]
+        struct BroadcastDim<const D1: usize, const D2: usize>;
```

Comment:
Why does the name switch to broadcast here?

Reply:
I should change it. The OP was named broadcast_to